zetasql/fuzzing/component/fuzz_targets/fuzz_target.h

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef ZETASQL_FUZZING_FUZZ_TARGET_H
+#define ZETASQL_FUZZING_FUZZ_TARGET_H
+
+#include <iostream>
+#include <string>
+
+namespace zetasql_fuzzer {
+
+class SQLStringArg;
+
+class FuzzTarget {
+ public:
+  // virtual void Visit(PositionalArg<zetasql::ParameterValueMap>& arg);
+  virtual void Visit(SQLStringArg& arg) { AbortVisit("SQLStringArg&"); }
+  virtual void Execute() = 0;
+
+ private:
+  virtual void AbortVisit(const std::string& type) {
+    std::cerr << "#Visit(" << type
```

Comment:
Consider using a logging method here (LOG(ERROR) or similar) over cerr.  This allows users to control the logging more uniformly.

Reply:
I believe we discussed about this before. It is not for production use because first, OSS-Fuzz may not export log file. And second, AbortVisit means something is apparently wrong, e.g., a required Visit function by a target object is not implemented, but is supplied in the test. This is a bug and we should notify the developer during development phase, and it should theoretically never invoked during production runtime. I don't see the value of using LOG here.

Reply:
Should I probably use ASSERT here instead?

Reply:
If you want this to log and then exit the program, consider LOG(FATAL).  See below comment about consistency with the rest of the library.